src/overlays/actors/ovl_En_Ice_Hono/z_en_ice_hono.c

```diff
@@ -21,37 +32,341 @@ const ActorInit En_Ice_Hono_InitVars = {
+    ICHAIN_F32(unk_4C, 60, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 400, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+static InitChainEntry sInitChainDroppedFlame[] = {
+    ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 400, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+static InitChainEntry sInitChainSmallFlame[] = {
+    ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 400, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+f32 EnIceHono_SquareDist(Vec3f* pos1, Vec3f* pos2) {
+    f32 dx = pos1->x - pos2->x;
+    f32 dz = pos1->z - pos2->z;
+
+    return SQ(dx) + SQ(dz);
+}
+
+void EnIceHono_InitCapturableFlame(Actor* thisx, GlobalContext* globalCtx) {
+    EnIceHono* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChainCapturableFlame);
+    Actor_SetScale(&this->actor, 0.0074f);
+    this->actor.flags |= 1;
+    Actor_SetHeight(&this->actor, 10.0f);
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInitCapturableFlame);
+    Collider_CylinderUpdate(&this->actor, &this->collider);
+
+    this->actor.colChkInfo.mass = 0xFF;
+    EnIceHono_SetupActionCapturableFlame(this);
+}
+
+void EnIceHono_InitDroppedFlame(Actor* thisx, GlobalContext* globalCtx) {
+    EnIceHono* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChainDroppedFlame);
+    this->actor.scale.x = this->actor.scale.z = this->actor.scale.y = 0.00002f;
+    this->actor.gravity = -0.3f;
+    this->actor.minVelocityY = -4.0f;
+    this->actor.shape.unk_08 = 0.0f;
+    this->actor.shape.rot.x = this->actor.shape.rot.y = this->actor.shape.rot.z = this->actor.posRot.rot.x =
+        this->actor.posRot.rot.y = this->actor.posRot.rot.z = 0;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInitDroppedFlame);
+    Collider_CylinderUpdate(&this->actor, &this->collider);
+
+    this->collider.dim.radius = this->actor.scale.x * 4000.4f;
+    this->collider.dim.height = this->actor.scale.y * 8000.2f;
+    this->actor.colChkInfo.mass = 253;
+    EnIceHono_SetupActionDroppedFlame(this);
+}
+
+void EnIceHono_InitSmallFlame(Actor* thisx, GlobalContext* globalCtx) {
+    EnIceHono* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChainSmallFlame);
+    this->actor.scale.x = this->actor.scale.z = this->actor.scale.y = 0.0008f;
+    this->actor.gravity = -0.3f;
+    this->actor.minVelocityY = -4.0f;
+    this->actor.shape.unk_08 = 0.0f;
+
+    EnIceHono_SetupActionSmallFlame(this);
+}
+
+void EnIceHono_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnIceHono* this = THIS;
+    s16 params = this->actor.params;
```

Comment:
If you remove the line you get regalloc differences and this was the only way I found to fix it.

Reply:
That doesn't really make sense since it's unused.
Have you tried using `thisx` or letting the permuter find something that makes more sense?

Reply:
just tried it out myself, `thisx` and `globalCtx2` didnt fix the issue.
you cant use the temp for usages of params either, so yeah something fishy is going on

Reply:
Yeah, it took me ages to get to that "solution". I also tried the permuter without success. When removing the line the compiler stores the actor params in `v1` instead of the "correct" `v0`. I don't get why it is doing that.

Reply:
@Roman971 label it as potential fake match and revisit?

Reply:
It's fine, I don't think marking it is necessary